### PR TITLE
[pre-commit] Add clang-format Google style.

### PR DESCRIPTION
**Pull request by olivier-stasse**
This PR add the pre-commit hook to check the C++ format. The format is following the Google style.
You need to use pre-commit to take advantage of it:
https://pre-commit.com/

Maybe this should be specified in the CONTRIBUTING.md file.

### CODE REVIEW

**Review by jcarpent — CHANGES_REQUESTED** [review]
It should be nice to add an ignore commit to clean the diff tree.
This has been done previously in hpp-fcl and eigenpy for reference.

**Comment by olivier-stasse** [comment]
@jcarpent Is this what you asked for ?

**Comment by jcarpent** [comment]
Something like https://github.com/Simple-Robotics/proxsuite/blob/main/.git-blame-ignore-revs

**Comment by olivier-stasse** [comment]
> Something like https://github.com/Simple-Robotics/proxsuite/blob/main/.git-blame-ignore-revs

Done. Thanks for the feedback.

File: bindings/python/module.cpp
Comment (andreadelprete) [inline bindings/python/module.cpp:23]: Are we sure that changing the order of the include isn't gonna affect the compilation?

Comment (olivier-stasse) [reply bindings/python/module.cpp:23]: Good point I forgot to put the flag to avoid sorting the includes. I will do it and will check more carefully the CI tests. I got trapped in thinking that the CI was failing because of osqp (first CI failure on this PR https://github.com/stack-of-tasks/tsid/actions/runs/4126738738/jobs/7129052586).

**Comment by olivier-stasse** [comment]
I think that it should be fine now. Solvers still fails so. But it is working on  my mac.

**Comment by jcarpent** [comment]
I've just provided a potential fix for ROS-CI, which has already been applied to Pinocchio and EigenPy. Let's see what happens.

**Comment by jcarpent** [comment]
Seems to work

**Review by jcarpent — APPROVED** [review]

**Comment by olivier-stasse** [comment]
Thanks @jcarpent for the fix and the approval. I am waiting for @andreadelprete approval.

**Comment by andreadelprete** [comment]
Looks fine to me. I'm gonna merge

**Pull request merged**
